ompi/tools/mpirun/main.c

```diff
@@ -154,23 +146,33 @@ int main(int argc, char *argv[])
 #endif
     }
 
-    full_prterun_path = find_prterun();
-    if (NULL == full_prterun_path) {
-        opal_show_help("help-mpirun.txt", "no-prterun-found", 1);
-        exit(1);
-    }
-
     /*
      * set environment variable for our install location
      * used within the OMPI prrte schizo component
      */
-
     setenv("OMPI_LIBDIR_LOC", opal_install_dirs.libdir, 1);
 
     // Set environment variable to tell PRTE what MCA prefixes belong
     // to Open MPI.
     setup_mca_prefixes();
 
+#if OMPI_HAVE_PRTE_LAUNCH
+
+    ret = prte_launch(argc, argv);
```

Comment:
As I said elsewhere, why not just use "prun_common"? Already exists, public symbol, does pretty much everything you have in your "prte_launch" code. The little bit that is left can easily be put here, just like we did with our "prte", and then call "prun_common". Avoid a bunch of code duplication that adds nothing.

Reply:
Well, you definitely don't want to emulate `prun` - you need to emulate `prterun`. `prun` is missing a bunch of stuff that you need for `mpirun`.

I haven't looked at `prte_launch` but can take a gander if you can point me at it. My guess is that we simply need to add a brief init function to PRRTE to fill in the gap. My concern is that you'll be adding maintenance burden for every time a new cmd line option gets added, or some other change internal to PRRTE requires modifying `prun_common`. Avoiding code duplication would probably be a good thing.

Reply:
Thank you sir!

I'm pretty brain-dead tonight; lemme look at this tomorrow...

Reply:
I don't think we're doing anything else with that header; that old `prte()` function listed in there might be stale...?